parser.y

```diff
@@ -3237,7 +3270,7 @@ Identifier:
 identifier | UnReservedKeyword | NotKeywordToken | TiDBKeyword
 
 UnReservedKeyword:
- "ACTION" | "ASCII" | "AUTO_INCREMENT" | "AFTER" | "ALWAYS" | "AVG" | "BEGIN" | "BIT" | "BOOL" | "BOOLEAN" | "BTREE" | "BYTE" | "CLEANUP" | "CHARSET"
+ "ACTION" | "ASCII" | "AUTO_INCREMENT" | "AFTER" | "ALWAYS" | "AVG" | "BEGIN" | "BIT" | "BOOL" | "BOOLEAN" | "BTREE" | "BYTE" | "CLEANUP" | "CHARSET" %prec charsetKwd
```

Comment:
Do we need to care about `"CHARACTER" "SET"`? I am not very clear about this.

Reply:
why we use `%prec` here?

Reply:
> Do we need to care about `"CHARACTER" "SET"`? I am not very clear about this.

@zimulala  The reason is `CHARACTER` is a reserved token but `CHARSET` is unreserved. For example, 

```
mysql> create database charset;
Query OK, 1 row affected

mysql> create database character;
(1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'character' at line 1")
```

Reply:
> why we use `%prec` here?

@zimulala @winkyao the reason is `CHARSET` is an unreserved keyword. There's a shift/reduce conflict when parsing `AlterDatabseStmt CHARSET`, YACC need to decide if charset is treated as `CHARACTER SET` keyword, or the database name.

BTW, this precedence leads to the following parsing result:

```
mysql> alter database charset=utf8;
(1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '=utf8' at line 1")

mysql> alter database character set=utf8;
Query OK, 1 row affected
```
And it is consistent with MySQL parser.